src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java

```diff
@@ -397,6 +397,21 @@ public long count(String parentKey, DetachedCriteria criteria) {
         return Transactions.<Long, DetachedCriteria>execute(dao.sessionFactory, true, dao::count, criteria);
     }
 
+    /**
+     * Queries using the specified criteria across all shards and returns the counts of rows satisfying the criteria.
+     * <b>Note:</b> This method runs the query serially and it's usage is not recommended.
+     * @param criteria The select criteria
+     * @return List of counts in each shard
+     */
+    public List<Long> count(DetachedCriteria criteria) {
```

Comment:
Move this to just abode `scatterGather` method and rename it to `countScatterGather`